components/SignInForm.tsx

```diff
@@ -0,0 +1,150 @@
+"use client";
+
+import Image from "next/image";
+import Link from "next/link";
+import { useRef, useState } from "react";
+
+// Logo Import
+import logo from "../public/Cacart_logo.png";
+import googlelogo from "../public/google_logo.png";
+
+type FormInputs = {
+  email: string;
+  password: string;
+};
+
+export default function SignInForm() {
+  // Set userData
+  const userData = useRef<FormInputs>({
```

Comment:
`useRef` is used to handle from inputs. Could you explain why `useRef` is used instead of `useState`? Any benefits?

Reply:
I think the main difference between `useRef` and `useState` is about how they cause re-renders. `useState` makes the component re-render every time the state changes, like when you type in a text field. But `useRef` doesn't do that. After looking into it more, I found that `useState` works better for when you need to keep track of changes right away, like checking form inputs as you type in our `SignUpForm`. So, I've updated our code to use `useState` for this purpose. This way, we make sure the form checks everything correctly as you fill it in. If I got anything wrong, or if you have more to add, please let me know. I'd appreciate any advice!

Reply:
Right on! There are clear differences between `useRef` and `useState` that React as a state machine interprets. As you've well said, we should use `useState` to offer real-time validation.